core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverJobBuilder.java

```diff
@@ -0,0 +1,83 @@
+package ai.timefold.solver.core.impl.solver;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.solver.SolverConfigOverride;
+import ai.timefold.solver.core.api.solver.SolverJob;
+import ai.timefold.solver.core.api.solver.SolverJobBuilder;
+
+/**
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <ProblemId_> the ID type of submitted problem, such as {@link Long} or {@link UUID}.
+ */
+public final class DefaultSolverJobBuilder<Solution_, ProblemId_> implements SolverJobBuilder<Solution_, ProblemId_> {
+
+    private final DefaultSolverManager<Solution_, ProblemId_> solverManager;
+    private ProblemId_ problemId;
+    private Function<? super ProblemId_, ? extends Solution_> problemFinder;
+    private Consumer<? super Solution_> bestSolutionConsumer;
+    private Consumer<? super Solution_> finalBestSolutionConsumer;
+    private BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler;
+    private SolverConfigOverride<Solution_> solverConfigOverride;
+
+    public DefaultSolverJobBuilder(DefaultSolverManager<Solution_, ProblemId_> solverManager) {
+        this.solverManager = Objects.requireNonNull(solverManager, "The SolverManager (" + solverManager + ") cannot be null.");
+        // The config is required by SolverFactory and it is always initialized
+        this.solverConfigOverride = new SolverConfigOverride<>();
```

Comment:
Personally, I'd create this instance when we need it. If the user doesn't provide one, create in on-demand in `run()`.